File: dashboard/src/types/subcategory.ts
// Subcategory types matching the real backend API (Spanish field names)

export type SubcategoryEstado = 'activo' | 'inactivo';

export interface Subcategory {
  id: number;
  nombre: string;
  imagen_url: string | null;
  categoria_id: number;
  categoria_nombre: string | null;
  orden: number;
  estado: SubcategoryEstado;
  productos_count: number;
  created_at: string;
  updated_at: string;
}

export interface SubcategoryCreate {
  nombre: string;
  categoria_id: number;
  imagen_url?: string | null;
  orden?: number;
  estado?: SubcategoryEstado;
}

export interface SubcategoryUpdate {
  nombre?: string;
  categoria_id?: number;
  imagen_url?: string | null;
  orden?: number;
  estado?: SubcategoryEstado;
}

File: rest_api/app/schemas/subcategory.py
"""Subcategory schemas — Spanish API field names."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field
from typing import Annotated


class SubcategoryRead(BaseModel):
    """Subcategory response shape matching spec section 3.5."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    nombre: str
    imagen_url: str | None = None
    categoria_id: int
    categoria_nombre: str
    orden: int = 0
    estado: str  # "activo" | "inactivo"
    productos_count: int = 0
    created_at: str
    updated_at: str


class SubcategoryCreate(BaseModel):
    """POST /api/v1/branches/{branch_id}/subcategories request body."""

    nombre: Annotated[str, Field(min_length=2, max_length=100)]
    categoria_id: int
    imagen_url: str | None = None
    orden: int | None = None
    estado: str = "activo"


class SubcategoryUpdate(BaseModel):
    """PUT request body — all fields optional."""

    nombre: Annotated[str | None, Field(min_length=2, max_length=100)] = None
    categoria_id: int | None = None
    imagen_url: str | None = None
    orden: int | None = None
    estado: str | None = None


class SubcategoryDeleteResponse(BaseModel):
    """Response for subcategory deletion with cascade counts."""

    message: str
    cascade: dict[str, int]
